demo/NostrSDKDemo/Views/KeyInputSectionView.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  KeyInputSectionView.swift
+//  NostrSDKDemo
+//
+//  Created by Joel Klabo on 8/13/23.
+//
+
+import SwiftUI
+import NostrSDK
+
+enum KeyType {
+    case `public`
+    case `private`
+}
+
+struct KeyInputSectionView: View {
+    
+    @Binding var key: String
+    @Binding var isValid: Bool
+
+    var type: KeyType
+
+    var body: some View {
+        HStack {
+            TextField(label(for: type),
+                      text: $key)
+                .font(.system(size: 16, weight: .regular, design: .monospaced))
+                .autocapitalization(.none)
+                .autocorrectionDisabled()
+
+            if key.isEmpty {
+                EmptyView()
+            } else if isValid {
+                Image(systemName: "checkmark.seal.fill")
+                    .foregroundColor(.green)
+            } else {
+                Image(systemName: "x.circle.fill")
+                    .foregroundColor(.red)
+            }
+        }
+        .onChange(of: key) { _ in
+            isValid = checkValidity(key: key, type: type)
+        }
+        .onAppear {
+            isValid = checkValidity(key: key, type: type)
+        }
+    }
+
+    private func label(for type: KeyType) -> String {
+        switch type {
+        case .public:
+            return "Public Key"
+        case .private:
+            return "Private Key"
+        }
+    }
+
+    private func checkValidity(key: String, type: KeyType) -> Bool {
+        switch type {
+        case .public:
+            return checkValidity(publicKey: key)
+        case .private:
+            return checkValidity(privateKey: key)
+        }
+    }
+
+    // swiftlint:disable unused_optional_binding
+    private func checkValidity(publicKey: String) -> Bool {
+        if publicKey.contains("npub") {
+            if let _ = PublicKey(npub: publicKey) {
```

Comment:
These 5 lines reduce to:
```swift
return PublicKey(pub: publicKey) != nil
```

Can be applied to all the other cases too.

Reply:
Nice, was able to get rid of this too: `    // swiftlint:disable unused_optional_binding
`